### ISSUE TITLE
Quartz throws runtime exceptions when application is published in single file mode

### ISSUE THREAD

**Issue by Kizari** [quartznet/quartznet#2403]
**Describe the bug**

When publishing an application that implements Quartz.NET as a single file executable (i.e. dependencies are embedded inside the executable), Quartz throws runtime exceptions. I am assuming this is due to dynamic loading of dependencies at runtime such as SQLite.

**Version used**

3.9.0.

**To Reproduce**

1. Create any kind of .NET application that will output an executable
2. Implement Quartz with SQLite persistence (not certain if only SQLite is affected)
3. Publish the application with "Produce Single File" box checked
4. Run the published executable and observe the crash noted under "Additional context" below

**Expected behavior**

The expected behaviour is that the application would be able to run as normal without throwing exceptions under these conditions.

**Additional context**
The following exception log was taken from Windows Event Viewer after following the reproduction steps:
```
Application: [redacted]
CoreCLR Version: 8.0.424.16909
.NET Version: 8.0.4
Description: The process was terminated due to an unhandled exception.
Exception Info: Quartz.SchedulerException: Database schema validation failed. Make sure you have created the database tables that Quartz requires using the database schema scripts. You can disable this check by setting quartz.jobStore.performSchemaValidation to false
 ---> Quartz.JobPersistenceException: Failed to obtain DB connection from data source 'default': System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.ArgumentNullException: Value cannot be null. (Parameter 'path1')
   at System.ArgumentNullException.Throw(String paramName)
   at System.IO.Path.Combine(String path1, String path2)
   at System.Data.SQLite.SQLiteConnection..ctor(String connectionString, Boolean parseViaFramework)
   at System.Data.SQLite.SQLiteConnection..ctor(String connectionString)
   at System.Data.SQLite.SQLiteConnection..ctor()
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
   --- End of inner exception stack trace ---
   at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
   at System.Reflection.ConstructorInfo.Invoke(Object[] parameters)
   at Quartz.Util.ObjectUtils.InstantiateType[T](Type type)
   at Quartz.Impl.AdoJobStore.Common.DbProvider.CreateConnection()
   at Quartz.Util.DBConnectionManager.GetConnection(String dataSourceName)
   at Quartz.Impl.AdoJobStore.JobStoreSupport.GetConnection()
 ---> System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.ArgumentNullException: Value cannot be null. (Parameter 'path1')
   at System.ArgumentNullException.Throw(String paramName)
   at System.IO.Path.Combine(String path1, String path2)
   at System.Data.SQLite.SQLiteConnection..ctor(String connectionString, Boolean parseViaFramework)
   at System.Data.SQLite.SQLiteConnection..ctor(String connectionString)
   at System.Data.SQLite.SQLiteConnection..ctor()
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
   --- End of inner exception stack trace ---
   at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
   at System.Reflection.ConstructorInfo.Invoke(Object[] parameters)
   at Quartz.Util.ObjectUtils.InstantiateType[T](Type type)
   at Quartz.Impl.AdoJobStore.Common.DbProvider.CreateConnection()
   at Quartz.Util.DBConnectionManager.GetConnection(String dataSourceName)
   at Quartz.Impl.AdoJobStore.JobStoreSupport.GetConnection()
   --- End of inner exception stack trace ---
   at Quartz.Impl.AdoJobStore.JobStoreSupport.GetConnection()
   at Quartz.Impl.AdoJobStore.JobStoreTX.GetNonManagedTXConnection()
   at Quartz.Impl.AdoJobStore.JobStoreSupport.ExecuteInNonManagedTXLock[T](String lockName, Func`2 txCallback, Func`3 txValidator, Nullable`1 requestorId, CancellationToken cancellationToken)
   at Quartz.Impl.AdoJobStore.JobStoreSupport.ExecuteInNonManagedTXLock[T](String lockName, Func`2 txCallback, Func`3 txValidator, Nullable`1 requestorId, CancellationToken cancellationToken)
   at Quartz.Impl.AdoJobStore.JobStoreSupport.Initialize(ITypeLoadHelper loadHelper, ISchedulerSignaler signaler, CancellationToken cancellationToken) [See nested exception: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.ArgumentNullException: Value cannot be null. (Parameter 'path1')
   at System.ArgumentNullException.Throw(String paramName)
   at System.IO.Path.Combine(String path1, String path2)
   at System.Data.SQLite.SQLiteConnection..ctor(String connectionString, Boolean parseViaFramework)
   at System.Data.SQLite.SQLiteConnection..ctor(String connectionString)
   at System.Data.SQLite.SQLiteConnection..ctor()
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
   --- End of inner exception stack trace ---
   at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
   at System.Reflection.ConstructorInfo.Invoke(Object[] parameters)
   at Quartz.Util.ObjectUtils.InstantiateType[T](Type type)
   at Quartz.Impl.AdoJobStore.Common.DbProvider.CreateConnection()
   at Quartz.Util.DBConnectionManager.GetConnection(String dataSourceName)
   at Quartz.Impl.AdoJobStore.JobStoreSupport.GetConnection()]
   --- End of inner exception stack trace ---
   at Quartz.Impl.AdoJobStore.JobStoreSupport.Initialize(ITypeLoadHelper loadHelper, ISchedulerSignaler signaler, CancellationToken cancellationToken)
   at Quartz.Impl.AdoJobStore.JobStoreTX.Initialize(ITypeLoadHelper loadHelper, ISchedulerSignaler signaler, CancellationToken cancellationToken)
   at Quartz.Impl.StdSchedulerFactory.Instantiate()
   at Quartz.Impl.StdSchedulerFactory.Instantiate()
   at Quartz.Impl.StdSchedulerFactory.GetScheduler(CancellationToken cancellationToken)
   at Quartz.ServiceCollectionSchedulerFactory.GetScheduler(CancellationToken cancellationToken)
   at Quartz.QuartzHostedService.StartAsync(CancellationToken cancellationToken)
   at Microsoft.Extensions.Hosting.Internal.Host.<StartAsync>b__15_1(IHostedService service, CancellationToken token)
   at Microsoft.Extensions.Hosting.Internal.Host.ForeachService[T](IEnumerable`1 services, CancellationToken token, Boolean concurrent, Boolean abortOnFirstException, List`1 exceptions, Func`3 operation)
   at Microsoft.Extensions.Hosting.Internal.Host.<StartAsync>g__LogAndRethrow|15_3(<>c__DisplayClass15_0&)
   at Microsoft.Extensions.Hosting.Internal.Host.StartAsync(CancellationToken cancellationToken)
   at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.Start(IHost host)
   at [redacted] [See nested exception: Quartz.JobPersistenceException: Failed to obtain DB connection from data source 'default': System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.ArgumentNullException: Value cannot be null. (Parameter 'path1')
   at System.ArgumentNullException.Throw(String paramName)
   at System.IO.Path.Combine(String path1, String path2)
   at System.Data.SQLite.SQLiteConnection..ctor(String connectionString, Boolean parseViaFramework)
   at System.Data.SQLite.SQLiteConnection..ctor(String connectionString)
   at System.Data.SQLite.SQLiteConnection..ctor()
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
   --- End of inner exception stack trace ---
   at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
   at System.Reflection.ConstructorInfo.Invoke(Object[] parameters)
   at Quartz.Util.ObjectUtils.InstantiateType[T](Type type)
   at Quartz.Impl.AdoJobStore.Common.DbProvider.CreateConnection()
   at Quartz.Util.DBConnectionManager.GetConnection(String dataSourceName)
   at Quartz.Impl.AdoJobStore.JobStoreSupport.GetConnection()
 ---> System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.ArgumentNullException: Value cannot be null. (Parameter 'path1')
   at System.ArgumentNullException.Throw(String paramName)
   at System.IO.Path.Combine(String path1, String path2)
   at System.Data.SQLite.SQLiteConnection..ctor(String connectionString, Boolean parseViaFramework)
   at System.Data.SQLite.SQLiteConnection..ctor(String connectionString)
   at System.Data.SQLite.SQLiteConnection..ctor()
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
   --- End of inner exception stack trace ---
   at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
   at System.Reflection.ConstructorInfo.Invoke(Object[] parameters)
   at Quartz.Util.ObjectUtils.InstantiateType[T](Type type)
   at Quartz.Impl.AdoJobStore.Common.DbProvider.CreateConnection()
   at Quartz.Util.DBConnectionManager.GetConnection(String dataSourceName)
   at Quartz.Impl.AdoJobStore.JobStoreSupport.GetConnection()
   --- End of inner exception stack trace ---
   at Quartz.Impl.AdoJobStore.JobStoreSupport.GetConnection()
   at Quartz.Impl.AdoJobStore.JobStoreTX.GetNonManagedTXConnection()
   at Quartz.Impl.AdoJobStore.JobStoreSupport.ExecuteInNonManagedTXLock[T](String lockName, Func`2 txCallback, Func`3 txValidator, Nullable`1 requestorId, CancellationToken cancellationToken)
   at Quartz.Impl.AdoJobStore.JobStoreSupport.ExecuteInNonManagedTXLock[T](String lockName, Func`2 txCallback, Func`3 txValidator, Nullable`1 requestorId, CancellationToken cancellationToken)
   at Quartz.Impl.AdoJobStore.JobStoreSupport.Initialize(ITypeLoadHelper loadHelper, ISchedulerSignaler signaler, CancellationToken cancellationToken) [See nested exception: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.ArgumentNullException: Value cannot be null. (Parameter 'path1')
   at System.ArgumentNullException.Throw(String paramName)
   at System.IO.Path.Combine(String path1, String path2)
   at System.Data.SQLite.SQLiteConnection..ctor(String connectionString, Boolean parseViaFramework)
   at System.Data.SQLite.SQLiteConnection..ctor(String connectionString)
   at System.Data.SQLite.SQLiteConnection..ctor()
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
   --- End of inner exception stack trace ---
   at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
   at System.Reflection.ConstructorInfo.Invoke(Object[] parameters)
   at Quartz.Util.ObjectUtils.InstantiateType[T](Type type)
   at Quartz.Impl.AdoJobStore.Common.DbProvider.CreateConnection()
   at Quartz.Util.DBConnectionManager.GetConnection(String dataSourceName)
   at Quartz.Impl.AdoJobStore.JobStoreSupport.GetConnection()]]
```

**Comment by lahma** [comment]
Can you create a PR improving the behavior? If not, I think it's easier just to close the issue as there's nobody to work to fix this.

**Comment by jafin** [comment]
@Kizari appears it may be a SQLite issue see https://stackoverflow.com/questions/76583021/value-cannot-be-null-parameter-path1-on-connectionstring
There is a response from a poster  `Looks like this will be fixed in 1.0.119`
Which version of SQLite are you using?

**Comment by Kizari** [comment]
@jafin My apologies, you're absolutely right. This is a problem with `System.Data.SQLite`, not Quartz. The issue is solved by downgrading `System.Data.SQLite` from version `1.0.118` to `1.0.117` as mentioned in your link. With this version, I am able to run the single file application fine with "Include native libraries for self extract" checked. Thank you kindly for the solution.

Closing this issue as it is not related to this repository.
